tests/test.py

```diff
@@ -143,6 +143,26 @@ def test_exception(self):
         except gp.GridPyError as e:
             self.assertEqual("Switch 'aa' not found", str(e))
 
+    def test_reduce_by_voltage(self):
+        n = gp.network.create_eurostag_tutorial_example1_network()
+        gp.loadflow.run_ac(n)
+        self.assertEqual(4, len(n.buses))
+        n.reduce(min=240, max=400)
+        self.assertEqual(2, len(n.buses))
+
+    def test_reduce_by_ids(self):
+        n = gp.network.create_eurostag_tutorial_example1_network()
+        gp.loadflow.run_ac(n)
+        self.assertEqual(4, len(n.buses))
+        n.reduce(ids=['P2'])
+        self.assertEqual(2, len(n.buses))
+
+    def test_reduce_by_subnetwork(self):
+        n = gp.network.create_eurostag_tutorial_example1_network()
+        gp.loadflow.run_ac(n)
+        self.assertEqual(4, len(n.buses))
+        n.reduce(vls=['VLGEN'], depths=[2])
```

Comment:
I don't think it is useful to support a list of vl/depth. Only one is enough. Also maybe in this case we should have an object to group voltage level id and depth as it has to be defined at the same time.

Reply:
I guess we can use a python tuple here for now : `vl_depth=('VL', 3)` ?

I think a list of VL + depth would be interesting but let's wait for a use case. Actually after thinking again I think the depth should be an option available for all predicates, for example a typical use case would be "my country + 4 substations around". In that case we could have a single option `depth=4`.

Reply:
ok so with a default value of 1 so that we don't need to check if depth is defined in case if vl_id defined?

Reply:
I think for now it's better to stick with the predicates in powsybl --> only add `vl_depth` tuple argument, and for all other arguments there is no depth.

But yes if we add later a generic depth argument, it could default to 1 (or 0 ?)

Reply:
Ok